pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -975,7 +973,17 @@ def process_circuits(
 
         pytket_pass = cast(Optional[BasePass], kwargs.get("pytketpass"))
 
-        language = cast(Language, kwargs.get("language", Language.QASM))
+        language: Optional[Language] = cast(Language, kwargs.get("language"))
+
+        if language is None:
+            language = Language.QASM
+            warnings.warn(
+                "The circuit is currently submitted as QASM by default. \
+The default is going to change. If you want to continue to use QASM \
+please add `language=Language.QASM` to the process_circuits function \
+to ensure the expected behavior. If you want to try QIR, please add \
```

Comment:
I'm not sure "expected behavior" is the right thing to say here: the behaviour should be indistinguishable, right?

Reply:
I was not really sure what to say, but there are some differences.
The main point are, that QIR can handle conditionals more efficient, but it might be hitting the register limit earlier.